hawkbit-security-core/src/main/java/org/eclipse/hawkbit/security/SystemSecurityContext.java

```diff
@@ -111,6 +117,44 @@ public SystemSecurityContext(final TenantAware tenantAware) {
         }
     }
 
+    /**
+     * Runs a given {@link Callable} within a system security context, which has
+     * the provided {@link GrantedAuthority}s to successfully run the
+     * {@link Callable}.
+     * 
+     * The security context will be switched to the a new
+     * {@link SecurityContext} and back after the callable is called.
+     * 
+     * @param tenant
+     *            to act as system code
+     * @param callable
+     *            to call within the security context
+     * @return the return value of the {@link Callable#call()} method.
+     */
+    // The callable API throws a Exception and not a specific one
+    @SuppressWarnings({ "squid:S2221", "squid:S00112" })
+    public <T> T runAsControllerAsTenant(@NotEmpty final String tenant, @NotNull final Callable<T> callable) {
+        final SecurityContext oldContext = SecurityContextHolder.getContext();
+        List<SimpleGrantedAuthority> authorities = Collections
+                .singletonList(new SimpleGrantedAuthority(SpringEvalExpressions.CONTROLLER_ROLE_ANONYMOUS));
+        try {
+            LOG.debug("entering system code execution");
```

Comment:
either remove the debug statement or fix its output ... "system code" is misleading here

Reply:
Had replicated the same log message like it is in runAsControllerAsTenant. If it is not required then I will remove it.